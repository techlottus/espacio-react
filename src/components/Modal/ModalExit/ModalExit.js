import React, { useEffect, useState } from "react";
import { Button } from "../../Button/Button";
import {
  btnModalBackExit,
  btnModalNextExit,
} from "../../../constants/Payment.constant";
import "./ModalExit.scss";
import { useSelector } from "react-redux";
import { testPaymenTitle } from "../../../constants/test/testPayment";

export const ModalExit = ({ isShow, onClose, onNext, flow }) => {
  const { texts } = useSelector((state) => state.texts);
  const [isModal, setIsModal] = useState(false);

  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]);

  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  const renderTitle = (typeFlow) => {
    switch (typeFlow) {
      case "procedures":
        return texts?.general.modals.modalExit.modalTitleProcedure;
      case "payment":
       return texts?.general.modals.modalExit.modalTitlePayment;
      default:
        return ''
    }
  };
  const handleModal = () => {
    return (
      <div className="modal-exit">
        <div className="modalcontent">
          <div className="modalheader">
            <span className="material-icons icon iconright">{texts?.general.modals.modalExit.icon}</span> 
            <div className="title">{renderTitle(flow)}</div>
          </div>
          <div className="modalbody">
            <div className="modalbtns">
              <div className="btnsecundary">
                <Button
                  data={{
                    ...btnModalBackExit,
                    title: texts?.general.modals.modalExit.cancelBtn,
                    isExpand: window.innerWidth < 991,
                    test: testPaymenTitle.btnModalExitCancel
                  }}
                  onClick={() => {
                    onClose();
                  }}
                />
              </div>
              <div className="btnprimary">
                <Button
                  data={{
                    ...btnModalNextExit,
                    title: texts?.general.modals.modalExit.exitBtn,
                    isExpand: window.innerWidth < 991,
                    test: testPaymenTitle.btnModalExitAccept
                  }}
                  onClick={() => {
                    onNext();
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
