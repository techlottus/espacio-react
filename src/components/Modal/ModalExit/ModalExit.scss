@import '../../../App.scss';
@import '../Modal.scss';

.modal-exit {
  @extend .modal-container;

  .modalcontent {

    .modalheader {
      justify-content: flex-start;
    }

  }

}