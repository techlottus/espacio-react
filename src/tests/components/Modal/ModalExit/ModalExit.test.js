import React from "react";
import { mount } from "enzyme";
import { ModalExit } from "../../../../components/Modal/ModalExit/ModalExit";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom/cjs/react-router-dom.min";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };

let initState = {
  texts: {},
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <BrowserRouter>
        <ModalExit history={historyMock} />
      </BrowserRouter>
    </Provider>
  );
};

describe("Test <ModalExit />", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
  describe("set isModal", () => {
    test("set isModal to true", () => {
      const wrapper = handleMount(initState);
    });
  });
});
