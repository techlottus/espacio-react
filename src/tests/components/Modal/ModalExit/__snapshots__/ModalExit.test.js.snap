// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ModalExit /> should init component 1`] = `""`;
